src/oidcendpoint/oauth2/authorization.py

```diff
@@ -507,10 +794,8 @@ def post_authentication(self, user, request, sid, **kwargs):
 
         _cookie = new_cookie(
             self.endpoint_context,
-            sub=user,
-            sid=sid,
+            sid=session_id,
             state=request["state"],
```

Comment:
```python
request.get("state")
```
state is optional

Reply:
state is mandatory in oidc, correct me if I'm wrong

Reply:
But not required. Hence if it's there in the request we add it to the cookie info otherwise it will just be ignored.